Gemfile

```diff
@@ -1,13 +1,15 @@
 source 'https://rubygems.org'
 
-ruby '2.5.3'
-gem 'puma'
+ruby '2.7.6'
+gem 'puma', '4.3.12'
 gem 'rails', '4.2.10'
 gem 'pg', '~> 0.18'
 gem 'sass-rails', '~> 5.0'
 gem 'uglifier', '>= 1.3.0'
 gem 'coffee-rails', '~> 4.1.0'
 
+gem 'json'
+gem 'bigdecimal', '1.3.5'
```

Comment:
These were added based on googling error messages after upgrading and trying to get into the rails console.